StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1651CodeFixProvider.cs

```diff
@@ -0,0 +1,90 @@
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    /// <summary>
+    /// Implements a code fix for <see cref="SA1651DoNotUsePlaceholderElements"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>To fix a violation of this rule, review the placeholder documentation for accuracy and remove the
+    /// &lt;placeholder&gt; tags.</para>
+    /// </remarks>
+    [ExportCodeFixProvider(nameof(SA1642SA1643CodeFixProvider), LanguageNames.CSharp)]
+    [Shared]
+    public class SA1651CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1651DoNotUsePlaceholderElements.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            // Require users review each removal of placeholder tags.
+            return null;
+        }
+
+        /// <inheritdoc/>
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!FixableDiagnostics.Contains(diagnostic.Id, StringComparer.Ordinal))
+                    continue;
+
+                var documentRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+                SyntaxNode syntax = documentRoot.FindNode(diagnostic.Location.SourceSpan, findInsideTrivia: true, getInnermostNodeForTie: true);
+                if (syntax == null)
+                    continue;
+
+                XmlElementSyntax xmlElementSyntax = syntax as XmlElementSyntax;
+                if (xmlElementSyntax == null)
+                {
+                    // We continue even for placeholders if they are empty elements (XmlEmptyElementSyntax)
+                    continue;
+                }
+
+                if (string.IsNullOrWhiteSpace(xmlElementSyntax.Content.ToString()))
+                {
+                    // The placeholder hasn't been updated yet.
+                    continue;
+                }
+
+                string description = "Finalize placeholder text";
+                context.RegisterCodeFix(CodeAction.Create(description, cancellationToken => this.GetTransformedDocumentAsync(context.Document, xmlElementSyntax, cancellationToken)), diagnostic);
+            }
+        }
+
+        private async Task<Document> GetTransformedDocumentAsync(Document document, XmlElementSyntax elementSyntax, CancellationToken cancellationToken)
+        {
+            SyntaxList<XmlNodeSyntax> content = elementSyntax.Content;
+            if (content.Count == 0)
+                return document;
+
+            var leadingTrivia = elementSyntax.StartTag.GetLeadingTrivia();
+            leadingTrivia = leadingTrivia.AddRange(elementSyntax.EndTag.GetTrailingTrivia());
```

Comment:
Should have been `StartTag`. However, I can't find any case where the trailing trivia of the start or end tag XML documentation node is not empty. If that holds true always this never would have revealed itself.